packages/perseus-editor/src/widgets/expression-editor.tsx

```diff
@@ -276,43 +277,44 @@ class ExpressionEditor extends React.Component<Props, State> {
 
     changeExpressionWidget: (
         index: number,
-        props: React.ComponentProps<typeof Expression>,
-    ) => void = (index, props) => {
+        input: PerseusExpressionUserInput,
+    ) => void = (index, input) => {
         const answerForm: AnswerForm = {
             ...this.props.answerForms[index],
-            value: props.value,
+            value: input,
         };
         this.updateAnswerForm(index, answerForm);
     };
 
     render(): React.ReactNode {
         const answerOptions: React.JSX.Element[] = this.props.answerForms.map(
             (ans: AnswerForm, index: number) => {
-                const expressionProps: React.ComponentProps<typeof Expression> =
-                    {
-                        // note we're using
-                        // *this.props*.{times,functions,buttonSets} since each
-                        // answer area has the same settings for those
-                        times: this.props.times,
-                        functions: this.props.functions,
-                        buttonSets: this.props.buttonSets,
-                        buttonsVisible: "focused",
-                        value: ans.value,
-                        // @ts-expect-error: Type '(props: React.ComponentProps<typeof Expression>) => void' is not assignable to type 'ChangeHandler'. Types of parameters 'props' and 'arg1' are incompatible.
-                        onChange: (
-                            props: React.ComponentProps<typeof Expression>,
-                        ) => this.changeExpressionWidget(index, props),
-                        trackInteraction: () => {},
-                        widgetId: this.props.widgetId + "-" + ans.key,
-                        visibleLabel: this.props.visibleLabel,
-                        ariaLabel: this.props.ariaLabel,
-                    } as const;
+                const expressionProps: Partial<
+                    React.ComponentProps<typeof Expression>
+                > = {
+                    // note we're using
+                    // *this.props*.{times,functions,buttonSets} since each
+                    // answer area has the same settings for those
+                    times: this.props.times,
+                    functions: this.props.functions,
+                    buttonSets: this.props.buttonSets,
+                    buttonsVisible: "focused",
+                    userInput: ans.value,
+                    // TODO: UniversalWidgetProps should have a generic type arg
+                    // to help scope what user input a widget consumes
+                    handleUserInput: ((input: PerseusExpressionUserInput) =>
+                        this.changeExpressionWidget(index, input)) as any,
```

Comment:
Is there a way to not use the casting to any here and on line 317? I'm still not the best at knowing when this is the right decision, so just wanted to check. I just try to avoid casting in general. XD

Reply:
Oh, lol, I thought you were talking about line 306 which it turns out I didn't need to cast for.

I don't know what to do about the cast on 317. We're expecting props for an Expression widget but the editor doesn't provide all of those props. This happens in a couple of places in the editors because the widgets in the editors don't need all of the stuff they need in the learner experience. For the most part this worked because of poor type coverage or use of `ts-expect-error` which is much worse than type casting.

I think this is still an improvement than what was here before and I think further work on this is beyond the scope of my current work.